log.go

```diff
@@ -80,6 +100,122 @@ type eventLogger struct {
 	// TODO add log-level
 }
 
+type activeSpanKeyType struct{}
+
+var activeSpanKey = activeSpanKeyType{}
+
+type activeSpan struct {
+	isFinished bool
+}
+
+func (el *eventLogger) Start(ctx context.Context, name string) context.Context {
+	el.Event(ctx, name, LoggableMap{"system": el.system})
+	as := &activeSpan{
+		isFinished: false,
+	}
+	ctx = context.WithValue(ctx, activeSpanKey, as)
+	span, ctx := opentrace.StartSpanFromContext(ctx, name)
+	span.SetTag("system", el.system)
+	return ctx
+}
+
+func (el *eventLogger) StartFromParentState(ctx context.Context, name string, parent []byte) (context.Context, error) {
+	sc, err := deserializeContext(parent)
+	if err != nil {
+		return nil, err
+	}
+	el.Event(ctx, name, LoggableMap{"system": el.system})
+
+	as := &activeSpan{
+		isFinished: false,
+	}
+	ctx = context.WithValue(ctx, activeSpanKey, as)
+	span, ctx := opentrace.StartSpanFromContext(ctx, name, otExt.RPCServerOption(sc))
+	span.SetTag("system", el.system)
+	return ctx, nil
+}
+
+func (el *eventLogger) SerializeContext(ctx context.Context) ([]byte, error) {
+	gTracer := opentrace.GlobalTracer()
+	var carrier bytes.Buffer
+	span := opentrace.SpanFromContext(ctx)
+	if err := gTracer.Inject(span.Context(), opentrace.Binary, carrier); err != nil {
+		return nil, errors.New("Failed to serialize context")
+	}
+	return carrier.Bytes(), nil
+}
+
+func (el *eventLogger) LogKV(ctx context.Context, key string, value interface{}) {
+	span := opentrace.SpanFromContext(ctx)
+	if span == nil {
+		_, file, line, _ := runtime.Caller(1)
+		log.Errorf("LogKV with no Span in context called on %s:%d", path.Base(file), line)
+		return
+	}
+	span.LogKV(key, value)
+}
+
+func (el *eventLogger) SetTag(ctx context.Context, key string, value interface{}) {
+	span := opentrace.SpanFromContext(ctx)
+	if span == nil {
+		_, file, line, _ := runtime.Caller(1)
+		log.Errorf("SetTag with no Span in context called on %s:%d", path.Base(file), line)
+		return
+	}
+	span.SetTag(key, value)
+}
+
+func (el *eventLogger) SetError(ctx context.Context, err error) {
+	if err == nil {
+		return
+	}
```

Comment:
I'd move this check down below the `span == nil` check (so we *always* log when calling `SetError` in the wrong places.